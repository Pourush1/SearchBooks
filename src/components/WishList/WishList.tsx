import React from 'react';
import { IBook } from '../../book-search/BookSearch';

interface IWishListProps {
  wishList: IBook[];
}

const WishList: React.FC<IWishListProps> = ({ wishList }) => {
  return (
    <div className="mb-4">
      <h4 className="m-auto py-3 text-center">
        My Reading Wishlist ({wishList.length})
      </h4>
      <div className="row" data-testid="bookList">
        {wishList.map((book: IBook, index: number) => {
          return (
            <div className="col-md-6 p-3" key={index} data-testid="book">
              <div className="card w-75">
                <img
                  className="card-img-top"
                  src={book.thumbnail}
                  alt="book cover"
                ></img>
                <p className="font-weight-bold">{book.title}</p>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default WishList;
