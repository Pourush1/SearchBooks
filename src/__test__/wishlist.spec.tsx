import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import WishList from '../components/WishList/WishList';

let documentBody: RenderResult;

describe('<Book />', () => {
  const props = {
    wishList: [
      {
        title: 'Java',
        published: '2016-11-05'
      },
      {
        title: 'Javascipt',
        published: '2017-11-05'
      }
    ]
  };

  beforeEach(() => {
    documentBody = render(<WishList {...props} />);
  });

  it('shows initial messages', () => {
    const myReadingList = screen.getByRole('heading');
    expect(myReadingList).toHaveTextContent('My Reading Wishlist');
  });

  it('renders correct number of books', () => {
    const bookList = screen.getAllByTestId('bookList');
    expect(bookList).toHaveLength(1);

    const book = screen.getAllByTestId('book');
    expect(book).toHaveLength(2);
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
