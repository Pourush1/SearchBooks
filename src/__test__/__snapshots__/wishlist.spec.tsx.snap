// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="mb-4"
    >
      <h4
        class="m-auto py-3 text-center"
      >
        My Reading Wishlist (
        2
        )
      </h4>
      <div
        class="row"
        data-testid="bookList"
      >
        <div
          class="col-md-6 pb-3"
          data-testid="book"
        >
          <div
            class="card w-75"
          >
            <img
              alt="book cover"
              class="card-img-top"
            />
            <p
              class="font-weight-bold"
            >
              Java
            </p>
          </div>
        </div>
        <div
          class="col-md-6 pb-3"
          data-testid="book"
        >
          <div
            class="card w-75"
          >
            <img
              alt="book cover"
              class="card-img-top"
            />
            <p
              class="font-weight-bold"
            >
              Javascipt
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
